app.go

```diff
@@ -418,12 +418,12 @@ func (h *Headscale) Serve() error {
 
 	err = h.ensureUnixSocketIsAbsent()
 	if err != nil {
-		panic(err)
+		return err
```

Comment:
I think this is sensible,

But could you wrap these errors to give the user more context, when we are first at improving :)

So:
```suggestion
		return fmt.Errorf("description: %w", err)
```